mwdb/web/src/components/Profile/Views/ProfileAPIKeys.js

```diff
@@ -187,10 +191,28 @@ export default function ProfileAPIKeys({ profile, updateProfile }) {
             <ConfirmationModal
                 isOpen={removeModalOpened}
                 onRequestClose={() => setRemoveModalOpened(false)}
-                onConfirm={(e) => removeApiKey(apiKeyToRemove)}
-                message={`Remove the API key ${apiKeyToRemove}?`}
+                onConfirm={(e) => removeApiKey(apiKeyToRemove.id)}
+                message={`Remove the API key ${apiKeyToRemove.name}?`}
                 confirmText="Remove"
             />
+            <ConfirmationModal
+                isOpen={apiKeyNameModalOpened}
+                onRequestClose={() => setApiKeyNameModalOpened(false)}
+                buttonStyle="btn-primary"
+                onConfirm={(e) => {
+                    setApiKeyNameModalOpened(false);
+                    const keyName = apiKeyNameRef.current.value;
+                    createApiKey(keyName);
+                }}
+                message={`Name of the API key`}
+                confirmText="Create"
+            >
+                <input
+                    className="form-control"
+                    ref={apiKeyNameRef}
```

Comment:
Uh I don't really like that ref here. Why not `useState`? (https://reactjs.org/docs/forms.html)

In addition, it would be nice to wrap that `<input>` with `<form>` so we can submit the new API key using Enter.
Great implementation of similar modal can be found here (`QuickQueryAddModal`): https://github.com/CERT-Polska/mwdb-core/blob/master/mwdb/web/src/components/RecentView/QuickQueryAddModal.js

Reply:
Ah yes, you're right. useState seems more fitting. I somehow missed that.

I'll wrap it with a form, too